earth_enterprise/src/common/khstl.h

```diff
@@ -358,6 +358,19 @@ makeset(const T &e1, const T &e2, const T &e3, const T &e4, const T &e5)
   tmp.insert(e5);
   return tmp;
 }
+template <class T>
+std::set<T>
+makeset(const T &e1, const T &e2, const T &e3, const T &e4, const T &e5, const T &e6)
```

Comment:
I'm not sure what you mean by refactoring the STL implementation. GCC supports initializer lists since 4.4, and the STL supports them, so you can create, e.g., a set initialized with whatever elements you want in a single line of code.

I see that you're being consistent with the code, I wouldn't hold it up for this particular question.

However, I think we should be getting away from this code, rather than expanding its use.  I would also like to make the version and type of compiler we require for building GEE explicit, so it's clear what code we can write.  I think all the OS distributions that we've listed as supported (RHEL 6, Ubuntu 14) have, at least, GCC 4.4, and GCC is now at version 7.